src/dbnode/persist/fs/types.go

```diff
@@ -700,10 +700,11 @@ type StreamedDataEntry struct {
 // StreamedMetadataEntry contains the metadata of single entry returned by streaming method.
 // The underlying data slices are reused and invalidated on every read.
 type StreamedMetadataEntry struct {
-	ID           ident.BytesID
-	EncodedTags  ts.EncodedTags
-	Length       int
-	DataChecksum uint32
+	ID            ident.BytesID
+	EncodedTags   ts.EncodedTags
+	Length        int
+	DataChecksum  uint32
+	IndexChecksum int64
```

Comment:
nit: any reason for having different data type (`int64` vs `uint32`) for `IndexChecksum`?

Reply:
Things are a bit weird there - the underlying structure uses int64, even though we use 32 bit values when calculating the actual checksums. Switched to `uint32` for `IndexChecksum`.